pkg/config/configuration.go

```diff
@@ -206,3 +276,272 @@ func containsKey(s []string, key string) bool {
 
 	return index < len(s) && s[index] == key
 }
+
+// ParseAccessControlSpec creates an in-memory copy of the Access Control Spec for fast lookup
+func ParseAccessControlSpec(accessControlSpec AccessControlSpec) (AccessControlList, error) {
+	var accessControlList AccessControlList
+	accessControlList.PolicySpec = make(map[string]AccessControlListPolicySpec)
+	accessControlList.DefaultAction = strings.ToLower(accessControlSpec.DefaultAction)
+
+	accessControlList.TrustDomain = accessControlSpec.TrustDomain
+	if accessControlSpec.TrustDomain == "" {
+		accessControlList.TrustDomain = DefaultTrustDomain
+	}
+
+	var invalidTrustDomain []string
+	var invalidNamespace []string
+	var invalidAppName bool
+	accessControlList.PolicySpec = make(map[string]AccessControlListPolicySpec)
+	for _, appPolicySpec := range accessControlSpec.AppPolicies {
+		invalid := false
+		if appPolicySpec.AppName == "" {
+			invalidAppName = true
+		}
+		if appPolicySpec.TrustDomain == "" {
+			invalidTrustDomain = append(invalidTrustDomain, appPolicySpec.AppName)
+			invalid = true
+		}
+		if appPolicySpec.Namespace == "" {
+			invalidNamespace = append(invalidNamespace, appPolicySpec.AppName)
+			invalid = true
+		}
+
+		if invalid || invalidAppName {
+			// An invalid config was found for this app. No need to continue parsing the spec for this app
+			continue
+		}
+
+		operationPolicy := make(map[string]AccessControlListOperationAction)
+
+		// Iterate over all the operations and create a map for fast lookup
+		for _, appPolicy := range appPolicySpec.AppOperationActions {
+			// The operation name might be specified as /invoke/*
+			// Store the prefix as the key and use the remainder as post fix for faster lookups
+			// Also, prepend "/" in case it is missing in the operation name
+			operation := appPolicy.Operation
+			if !strings.HasPrefix(operation, "/") {
+				operation = "/" + operation
```

Comment:
operation might not always be an HTTP RESTful operation, like `/orders`. In case of a gRPC app, it will likely be something like `Orders` and will never have a `/` prefix.

Reply:
The code will handle that case